pkg/algorithm/nodeselect/v1alpha2/select_node.go

```diff
@@ -0,0 +1,186 @@
+/*
+Copyright 2018 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha2
+
+import (
+	"github.com/golang/glog"
+	ndmapis "github.com/openebs/maya/pkg/apis/openebs.io/ndm/v1alpha1"
+	apis "github.com/openebs/maya/pkg/apis/openebs.io/v1alpha1"
+	bd "github.com/openebs/maya/pkg/blockdevice/v1alpha2"
+	bdc "github.com/openebs/maya/pkg/blockdeviceclaim/v1alpha1"
+	csp "github.com/openebs/maya/pkg/cstor/newpool/v1alpha3"
+	nodeapis "github.com/openebs/maya/pkg/kubernetes/node/v1alpha1"
+	"github.com/openebs/maya/pkg/volume"
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// SelectNode returns a node where pool should be created.
+func (ac *Config) SelectNode() (*apis.PoolSpec, string, error) {
+	usedNodes, err := ac.GetUsedNode()
+	if err != nil {
+		return nil, "", errors.Wrapf(err, "could not get used nodes list for pool creation")
+	}
+	for _, pool := range ac.CSPC.Spec.Pools {
+		// pin it
+		pool := pool
+		nodeName, err := ac.GetNodeFromLabelSelector(pool.NodeSelector)
+		if err != nil {
+			glog.Errorf("could not use node for selectors {%v}", pool.NodeSelector)
+			continue
+		}
+		if !usedNodes[nodeName] {
+			return &pool, nodeName, nil
+		}
+	}
+	return nil, "", errors.New("no node qualified for pool creation")
+}
+
+// GetNodeFromLabelSelector returns the node name selected by provided labels
+func (ac *Config) GetNodeFromLabelSelector(labels map[string]string) (string, error) {
+	nodeList, err := nodeapis.NewKubeClient().List(metav1.ListOptions{LabelSelector: getLabelSelectorString(labels)})
+	if err != nil {
+		return "", errors.Wrap(err, "failed to get node list from the node selector")
+	}
+	if len(nodeList.Items) != 1 {
+		return "", errors.Errorf("could not get a unique node from the given node selectors")
+	}
+	if !nodeapis.NewBuilder().WithAPINode(&nodeList.Items[0]).Node.IsReady() {
+		return "", errors.Errorf("node {%s} is not ready", nodeList.Items[0].Name)
+	}
+	return nodeList.Items[0].Name, nil
```

Comment:
can we return error If we get more than one node?(we can't believe input from user assume the case where user didn't provide unique labels on node)

Reply:
Already done